gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteHasNoDuplicateStopsValidatorTest.java

```diff
@@ -0,0 +1,191 @@
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    RouteHasNoDuplicateStopsValidator validator = new RouteHasNoDuplicateStopsValidator();
+
+    // Create tripTable:
+    validator.tripTable =
+        GtfsTripTableContainer.forEntities(Arrays.asList(trip1_route1, trip2_route1), noticeContainer);
+
+    // Create stopTimeTable:
+    validator.stopTimeTable = GtfsStopTimeTableContainer.forEntities(
+        Arrays.asList(stopTimeTrip1_Stop1, stopTimeTrip1_Stop2, stopTimeTrip2_Stop1, stopTimeTrip2_Stop2), noticeContainer);
+
+    // Create stopTable:
+    validator.stopTable =
+        GtfsStopTableContainer.forEntities(Arrays.asList(stopId1001, stopId1002), noticeContainer);
+    
+    validator.validate(noticeContainer);
+    System.out.println(noticeContainer.getNotices());
+    assertThat(noticeContainer.getNotices())
+        .containsExactly(new RouteWithDuplicateStopNotice(/* stopName = */ "stop1",
+            /* prevStopId  = */ "1001", /* prevCsvRowNumber = */ 3, /* stopId  = */ "1002",
+            /* csvRowNumber = */ 4, /* routeId = */ "route1"));
+  }
+
+  @Test
+  public void routeNoDuplicatesShouldNotGenerateNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    RouteHasNoDuplicateStopsValidator validator = new RouteHasNoDuplicateStopsValidator();
+
+    // Create tripTable:
+    validator.tripTable =
+        GtfsTripTableContainer.forEntities(Arrays.asList(trip1_route1), noticeContainer);
+
+    // Create stopTimeTable:
+    validator.stopTimeTable = GtfsStopTimeTableContainer.forEntities(
+        Arrays.asList(stopTimeTrip1_Stop2, stopTimeTrip1_Stop3), noticeContainer);
+
+    // Create stopTable:
+    validator.stopTable =
+        GtfsStopTableContainer.forEntities(Arrays.asList(stopId1002, stopId1003), noticeContainer);
+    
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices()).isEmpty();
+  }
+  
+  @Test
+  public void shouldGenerateNoticeForEachRepeatedDuplicateOnEachRoute() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    RouteHasNoDuplicateStopsValidator validator = new RouteHasNoDuplicateStopsValidator();
+
+    // Create tripTable:
+    validator.tripTable =
+        GtfsTripTableContainer.forEntities(Arrays.asList(trip1_route1, trip2_route1, trip3_route2), noticeContainer);
+
+    // Create stopTimeTable:
+    validator.stopTimeTable = GtfsStopTimeTableContainer.forEntities(
+        Arrays.asList(stopTimeTrip1_Stop1, stopTimeTrip1_Stop2, stopTimeTrip2_Stop1, stopTimeTrip2_Stop2, stopTimeTrip3_Stop1, stopTimeTrip3_Stop2), noticeContainer);
+
+    // Create stopTable:
+    validator.stopTable =
+        GtfsStopTableContainer.forEntities(Arrays.asList(stopId1001, stopId1002), noticeContainer);
+    
+    validator.validate(noticeContainer);
+    System.out.println(noticeContainer.getNotices().size());
+    System.out.println(noticeContainer.getNotices());
```

Comment:
Same as above with the printing. Remove them if you don't need to print.

Reply:
Removed :)